pkg/controller/postgres/postgres_controller.go

```diff
@@ -194,6 +203,8 @@ func (r *ReconcilePostgres) Reconcile(request reconcile.Request) (reconcile.Resu
 			return reconcile.Result{Requeue: true, RequeueAfter: p.GetReconcileTime(instance)}, nil
 		}
 
+		annotations.Add(instance, annotations.Created, annotations.Created)
```

Comment:
Yes, I would like to see a mapping here of the resource ID returned from AWS to the CR.
Not sure if this is in the scope of your current ticket, and I would not like to block the PR by adding this change. I will leave it up to yourself, depending on how much work it would take. 

Another option would be creating a JIRA for this, as you have laid the foundation here

Reply:
I'll leave that for now: currently all of this logic is at the top level of the controllers, so that we wouldn't have to think about it when adding future providers. I'm not sure what provider-specific info is available at this level. E.g. I don't know how to get the AWS resource ID here, or whatever the equivalent identifier for an in-cluster postgresql would be is.